tools/etcd-dump-logs/etcd-dump-log_test.go

```diff
@@ -0,0 +1,217 @@
+// Copyright 2018 The etcd Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"bytes"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path"
+	"path/filepath"
+	"strings"
+	"testing"
+
+	"github.com/coreos/etcd/etcdserver/etcdserverpb"
+	"github.com/coreos/etcd/pkg/pbutil"
+	"github.com/coreos/etcd/raft/raftpb"
+	"github.com/coreos/etcd/wal"
+	"go.uber.org/zap"
+)
+
+func TestEtcdDumpLogEntryType(t *testing.T) {
```

Comment:
Check at the beginning of tests, if `etcd-dump-logs` exists?

See https://github.com/coreos/etcd/blob/62a385b1ca2560847ec5b82ec0b40253320fd4e1/tests/e2e/etcd_release_upgrade_test.go#L32-L35